next permutation in c++

```diff
@@ -0,0 +1,69 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+//to print a vector
+void print(vector<int> a,int n){ 
+	for(int i=0;i<n;i++)
+		cout<<a[i]<<" ";
+	cout<<endl;
```

Comment:
You should use `std::cout << "\n"` instead, as `std::endl` will flush the stream, making it slower than necessary.